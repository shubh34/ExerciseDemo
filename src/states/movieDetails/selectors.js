import get from 'lodash/get';
import { createSelector } from 'reselect';

const getMovieDetails = state => get(state, 'movieDetails', '');

export const getHighLightedScore = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.scores', []).find(score => score.highlighted));

export const getMovieTitle = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.title', ''));

export const getSnapshotImageUrl = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.images.snapshot', '').replace('.jpeg', '-width1920-quality80.jpeg'));

export const getMovieDescription = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.plot', ''));

export const getScore = createSelector(getHighLightedScore, scores => get(scores, 'score', ''));

export const getVotes = createSelector(getHighLightedScore, scores => get(scores, 'formatted_amount_of_votes', ''));
