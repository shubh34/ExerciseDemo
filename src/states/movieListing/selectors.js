import get from 'lodash/get';

export const getMovieListHeader = (state, id) => get(state, `movieListing[${id}].data.name`);

export const getMovieListByCategory = (state, id) => get(state, `movieListing[${id}].data.contents.data`, []).map(movie => ({
	artwork: movie.images.artwork.replace('.jpeg', '-width217-quality80.jpeg'),
	id: movie.id,
	score: movie.highlighted_score.score,
	votes: movie.highlighted_score.formatted_amount_of_votes,
}));
