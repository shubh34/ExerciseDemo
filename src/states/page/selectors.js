export const getCurrentPage = state => state.page.name;
export const getBackPage = state => state.page.backPage;
