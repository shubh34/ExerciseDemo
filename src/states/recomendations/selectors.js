import get from 'lodash/get';
import { createSelector } from 'reselect';

export const getRecommendedMovies = state => get(state, 'recomendations.data', []);

export const getFormattedRecomendedMovieDetails = createSelector(
	getRecommendedMovies,
	movies => movies.map(movieDetails => ({
		id: movieDetails.id,
		artwork: movieDetails.images.artwork.replace('.jpeg', '-width217-quality80.jpeg'),
		score: movieDetails.highlighted_score.score,
		votes: movieDetails.highlighted_score.formatted_amount_of_votes
	}))
);
