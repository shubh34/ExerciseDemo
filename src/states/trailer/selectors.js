import get from 'lodash/get';

export const getMovieTrailerUrl = state => get(state, 'trailer.trailer.data.stream_infos[0].url');
