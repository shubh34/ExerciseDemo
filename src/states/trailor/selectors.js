import get from 'lodash/get';

export const getMovieTrailorUrl = state => get(state, 'trailor.trailor.data.stream_infos[0].url');
