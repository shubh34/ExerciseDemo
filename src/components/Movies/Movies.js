import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { fetchMovieListByCategory } from '../../states/movieListing/action';
import SectionHeader from '../../sharedComponents/SectionHeader/SectionHeader';
import Slider from '../../sharedComponents/Slider/Slider';
import SliderItem from '../../sharedComponents/Slider/SliderItem/SliderItem';
import { getMovieListHeader, getMovieListByCategory } from '../../states/movieListing/selectors';


const mapDispatch = dispatch => ({
	fetchMovieList: id => dispatch(fetchMovieListByCategory(id)),
});
const mapState = (state, props) => ({
	movieCategoryHeader: getMovieListHeader(state, props.id),
	movieList: getMovieListByCategory(state, props.id),
});
export const Movies = (props) => {
	const {
		id, fetchMovieList, movieCategoryHeader, movieList,
	} = props;
	useEffect(() => {
		fetchMovieList(id);
	}, [id]);
	return (
		<div className="movies">
			<SectionHeader header={movieCategoryHeader} />
			<Slider>
				{movieList.map(
					({
						artwork, score, id, votes,
					}, index) => (
						<SliderItem movie={artwork} score={score} id={id} votes={votes} key={index} />
					),
				)}
			</Slider>
		</div>
	);
};

Movies.propTypes = {
	id: PropTypes.string.isRequired,
	movieCategoryHeader: PropTypes.string.isRequired,
	movieList: PropTypes.arrayOf(PropTypes.shape({
		artwork: PropTypes.string.isRequired,
		score: PropTypes.number.isRequired,
		id: PropTypes.string.isRequired,
		votes: PropTypes.string.isRequired,
	})),
	fetchMovieList: PropTypes.func.isRequired,
};
export default connect(mapState, mapDispatch)(Movies);
