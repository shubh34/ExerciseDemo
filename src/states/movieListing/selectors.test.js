import { getMovieListByCategory, getMovieListHeader } from './selectors';
import { rawMovies } from '../../../tests-stub/Movies';

describe('MovieListing Selector', () => {
	it('should return Movie list header', () => {
		const state = { movieListing: rawMovies };
		expect(getMovieListHeader(state, 'populares-en-taquilla')).toMatchSnapshot();
	});
	it('should return Movie list ', () => {
		const state = { movieListing: rawMovies };
		expect(getMovieListByCategory(state, 'populares-en-taquilla')).toMatchSnapshot();
	});

	it('should return Movie list as empty when category is not in state ', () => {
		const state = { movieListing: rawMovies };
		expect(getMovieListByCategory(state, 'populares-en-taquillhjk')).toEqual([]);
	});
});
