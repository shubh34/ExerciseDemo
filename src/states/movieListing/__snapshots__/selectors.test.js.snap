// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListing Selector should return Movie list  1`] = `
Array [
  Object {
    "artwork": "https://images-2.wuaki.tv/system/artworks/86665/master/capitana-marvel-1560352921-width217-quality80.jpeg",
    "id": "capitana-marvel",
    "score": 7,
    "votes": "6,5K",
  },
  Object {
    "artwork": "https://images-3.wuaki.tv/system/artworks/54649/master/a-47-metros-1558460397-width217-quality80.jpeg",
    "id": "a-47-metros",
    "score": 5.6,
    "votes": "1,6K",
  },
  Object {
    "artwork": "https://images-2.wuaki.tv/system/artworks/86659/master/green-book-1558483188-width217-quality80.jpeg",
    "id": "green-book",
    "score": 8.3,
    "votes": "4K",
  },
  Object {
    "artwork": "https://images-2.wuaki.tv/system/artworks/43779/master/mascotas-1558452275-width217-quality80.jpeg",
    "id": "mascotas",
    "score": 6.1,
    "votes": "5,6K",
  },
  Object {
    "artwork": "https://images-2.wuaki.tv/system/artworks/89878/master/mula-1558485497-width217-quality80.jpeg",
    "id": "mula",
    "score": 6.5,
    "votes": "1,8K",
  },
  Object {
    "artwork": "https://images-0.wuaki.tv/system/artworks/89879/master/escape-room-2019-1559899512-width217-quality80.jpeg",
    "id": "escape-room-2019",
    "score": 6.4,
    "votes": "1,5K",
  },
  Object {
    "artwork": "https://images-1.wuaki.tv/system/artworks/86635/master/perdiendo-el-este-1559228752-width217-quality80.jpeg",
    "id": "perdiendo-el-este",
    "score": 4.2,
    "votes": "20",
  },
  Object {
    "artwork": "https://images-3.wuaki.tv/system/artworks/79289/master/creed-ii-la-leyenda-de-rocky-1558478964-width217-quality80.jpeg",
    "id": "creed-ii-la-leyenda-de-rocky",
    "score": 6.7,
    "votes": "2,3K",
  },
]
`;

exports[`MovieListing Selector should return Movie list header 1`] = `"Películas más vistas en Alquiler"`;
